server/mux.go

```diff
@@ -18,20 +18,21 @@ type mux struct {
 func New(options ...Option) *mux {
 	ss := &mux{
 		router: &router{
-			prefix:     "",
-			mux:        http.NewServeMux(),
+			prefix: "",
+			mux:    http.NewServeMux(),
+			baseMiddleware: []Middleware{
```

Comment:
@tatang26 should we have this list of middleware somewhere else?

Reply:
I mean at the package level, so we don't have to carry this on every router.

Reply:
@paganotoni Sure, I can do this